File: Intro/CS50/pset6/caesar.py
import sys

# check for proper usage
if len(sys.argv) != 2:
    print ("Usage: python caesar.py k")
    raise SystemExit

# get text in
print("plaintext:  ", end="")
in_word = input()
out_word = ""
k = int(sys.argv[1])
for i in range(0, len(in_word)):
    # for uppercase letters
    if str.isupper(in_word[i]) == True:
        out_word += chr((ord(in_word[i]) - 65 + k) % 26 + 65)
    # for lowercase letters
    elif str.islower(in_word[i]) == True:
        out_word += chr((ord(in_word[i]) - 97 + k) % 26 + 97)
    # for non-alpha characters
    else:
        out_word += in_word[i]
print("ciphertext: {}".format(out_word))

File: Intro/CS50/pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, string argv[]){
    if(argc != 2){
        printf("Usage: ./caesar k\n");
        return 1;
    };
    int k = atoi(argv[1]);
    if(k < 0){
        printf("Usage: ./caesar k\n");
        return 0;
    };
    printf("plaintext:\t");
    string str = get_string();
    k %= 26;
    int p = 0;
    for(int i = 0; i < strlen(str); i++){
        if(isalpha(str[i])){
            if(isupper(str[i])){
                p = str[i] + k;
                if(p > 90){
                    p -= 26;
                };
                str[i] = p;
            }else{
                p = str[i] + k;
                if(p > 122){
                    p -= 26;
                };
                str[i] = p;
            };
        };
    };
    printf("ciphertext:\t%s\n", str);
    return 0;
}
